### PR TITLE
Test on Julia 1.3, drop support for Julia 0.7, bump version to 0.4.0

### PR DESCRIPTION
**Pull request by DilumAluthge**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -3,16 +3,16 @@ os:
   - linux
   - osx
 julia:
-  - 0.7
   - 1.0
   - 1.1
   - 1.2
+  - 1.3
```
Comment (timholy) [inline .travis.yml]: How about changing this to `1` so that it always tracks the latest stable release?

Comment (DilumAluthge) [reply .travis.yml]: Done!

File: Project.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 name = "AxisArrays"
 uuid = "39de3d68-74b9-583c-8d2d-e117c070f3a9"
-version = "0.3.3"
+version = "0.3.4"
```
Comment (timholy) [inline Project.toml]: I was about to hit merge and then remembered, are we allowed to narrow the range of supported Julia versions on a patch release?

Comment (DilumAluthge) [reply Project.toml]: You are correct :( we are not. I'll bump to 0.4.0.

Comment (DilumAluthge) [reply Project.toml]: Done.

**Comment by timholy** [comment]
Thanks!

**Pull request merged**